### DISCUSSION TITLE
Bbox or dimensions of port

### DISCUSSION THREAD

**Discussion by grafandreas** [clientIO/joint#2006] [Q&A]
Dear all,

is it possible to get the bbox (or at least dimensions) of a port?

Regards,

Andreas

**Comment by kumilingus** [comment]
It depends. What are you trying to do?
Do you need the size of port itself? Or including the label?

```js
// port position (as calculated by port layout) and angle
element.getPortsPositions(portGroup)[portId];
// you should know the size of the port without label (you are the one who defined the port or it's 20x20 by default)
```

```js
// position relative to the port position determined by port layout
elementView.getNodeBoundingRect(elementView.findPortNode(portId, 'root'));
// absolute position and size
elementView.getNodeBBox(elementView.findPortNode(portId, 'root'));
```

**Reply by grafandreas** [reply]
I have a port with an embedded label (such as in the Chatbot example), so the size of the port is not fixed.

**Reply by kumilingus** [reply]
If the port size depends on the label inside, then you can inspire [here](https://codepen.io/jointjs/pen/yLKGgaR). There is a method (`measureText()`) to measure the size of the port label before it's rendered on paper.
